api/src/main/java/org/openmrs/module/commonreports/reports/DRCCD4MonitoringReportManager.java

```diff
@@ -0,0 +1,281 @@
+import org.openmrs.module.reporting.report.manager.ReportManagerUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.openmrs.module.reporting.cohort.definition.BaseObsCohortDefinition.TimeModifier;
+import org.openmrs.module.reporting.cohort.definition.AgeCohortDefinition;
+import org.openmrs.module.reporting.common.DateUtil;
+
+@Component
+public class DRCCD4MonitoringReportManager extends ActivatedReportManager {
+	
+	@Autowired
+	private InitializerService inizService;
+	
+	@Override
+	public boolean isActivated() {
+		//return inizService.getBooleanFromKey("report.drc.hivStage.active", false);
+		return true;
+		
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0.0-SNAPSHOT";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "45631da8-1e16-414f-9894-4aafad370c77";
+	}
+	
+	@Override
+	public String getName() {
+		return MessageUtil.translate("commonreports.report.drc.cd4Monitoring.reportName");
+	}
+	
+	@Override
+	public String getDescription() {
+		return MessageUtil.translate("commonreports.report.drc.cd4Monitoring.reportDescription");
+	}
+	
+	private Parameter getStartDateParameter() {
+		return new Parameter("startDate", MessageUtil.translate("commonreports.report.util.reportingStartDate"), Date.class);
+	}
+	
+	private Parameter getEndDateParameter() {
+		return new Parameter("endDate", MessageUtil.translate("commonreports.report.util.reportingEndDate"), Date.class);
+	}
+	
+	public static String col1 = "";
+	
+	public static String col2 = "";
+	
+	public static String col3 = "";
+	
+	public static String col4 = "";
+	
+	public static String col5 = "";
+	
+	public static String col6 = "";
+	
+	public static String col7 = "";
+	
+	public static String col8 = "";
+	
+	public static String col9 = "";
+	
+	public static String col10 = "";
+	
+	public static String col11 = "";
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> params = new ArrayList<Parameter>();
+		params.add(getStartDateParameter());
+		params.add(getEndDateParameter());
+		return params;
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		
+		// cd4Monitoring Grouping
+		CohortCrossTabDataSetDefinition cd4Monitoring = new CohortCrossTabDataSetDefinition();
+		cd4Monitoring.addParameters(getParameters());
+		rd.addDataSetDefinition(getName(), Mapped.mapStraightThrough(cd4Monitoring));
+		
+		Map<String, Object> parameterMappings = new HashMap<String, Object>();
+		parameterMappings.put("onOrAfter", "${startDate}");
+		parameterMappings.put("onOrBefore", "${endDate}");
+		SqlCohortDefinition sqd = new SqlCohortDefinition();
+		
+		// CD4 value in range
```

Comment:
This is more of CD4 result in range right?

Reply:
Thanks. Fixed it. I had initially relied on the obs date of the cd4 results instead due to the possible complexity for the second query. Fix this now